src/backend/distributed/deparser/ruleutils_15.c

```diff
@@ -1563,8 +1563,12 @@ set_join_column_names(deparse_namespace *dpns, RangeTblEntry *rte,
 
 	/* Assert we processed the right number of columns */
 #ifdef USE_ASSERT_CHECKING
-	while (i < colinfo->num_cols && colinfo->colnames[i] == NULL)
```

Comment:
The premise of this assertion is coming from the comments section of the definition of `deparse_columns`  - **Entries with `is_new_col` false must match the non-NULL colnames entries one-for-one.**

However in the actual code, we are setting the colnames to the actual names even when the `is_new_col` is `true`

```
set_relation_column_names()
	For each column - ncolumns = tupdesc->natts;
		colinfo->colnames is set irrespective of the below flag
		         /* And mark them as new or not */
                colinfo->is_new_col[j] = (i >= noldcolumns);
```

Reply:
By the way, this assertion failure is unrelated to the current fix. However, the bug is masking this assert-failure. As we addressed the issue, the code proceeded and encountered this failure. If we remove our fix and comment out `AcquireRewriteLocks()`, we will encounter this assertion in the current `main` code line.

Reply:
Yes, they should.